### PR TITLE
Add-On shop updates

### PR DESCRIPTION
**Pull request by zbtirrell**
🎟 [TCMN-70]
---
A bit of pre-warning for whoever does this code review: this is going to deviate a bit from standards… Yeah, I know, I should be modeling best behavior and the bastions of all things quality here. 

But! The App Shop is _rarely_ updated - and almost always by me personally. So, in the attempt at speed and efficiency, this is going to cut a lot of corners. 

I still want a code review - and we should still fix anything egregious, but, (for example) there's no BEM here, and we're all just going to be OK with that this one time.

[TCMN-70]: https://moderntribe.atlassian.net/browse/TCMN-70

### CODE REVIEW

File: src/admin-views/app-shop.php [hunk 1]
```diff
@@ -1,95 +1,223 @@
...
+				?>
+			</div>
+		</div>
+	</div>
+
+	<div id="tribe-extensions" class="tribe-content">
+		<div class="content-wrapper">
+			<div class="addon-grid">
+				<h2><?php _e( 'Extensions Headline', 'tribe-common' );?></h2>
+				<p><?php _e( 'Some text about why extensions are great', 'tribe-common' ); ?></p>
```
Comment (sc0ttkclark) [inline src/admin-views/app-shop.php:178]:
```suggestion
				<h2><?php esc_html_e( 'Extensions Headline', 'tribe-common' );?></h2>
				<p><?php esc_html_e( 'Some text about why extensions are great', 'tribe-common' ); ?></p>
```

File: src/admin-views/app-shop.php [hunk 2]
```diff
@@ -1,95 +1,223 @@
...
 								}
 								?>
-							</div>
+							</ul>
+						</div>
 
-							<a class="button <?php esc_attr_e( $button_class ); ?>" href="<?php echo esc_url( $product->link ); ?>"><?php echo $button_label; // escaped above ?></a>
+						<div class="cta narrow">
+							<a class="button" href="<?php echo esc_url( $bundle->link ); ?>"><?php esc_html_e( 'Save With A Bundle', 'tribe-common' ); ?></a>
+							<span class="discount"><?php echo esc_html( $bundle->discount );?></span>
```
Comment (sc0ttkclark) [inline src/admin-views/app-shop.php:163]:
```suggestion
							<span class="discount"><?php echo esc_html( $bundle->discount ); ?></span>
```

File: src/admin-views/app-shop.php [hunk 3]
```diff
@@ -1,95 +1,223 @@
...
+			</div>
+		</div>
+	</div>
 
-				foreach ( $products as $i => $product ) {
+	<div id="tribe-bundles" class="tribe-content">
+		<div class="content-wrapper">
+			<div class="addon-grid">
+				<h2><?php _e( 'Get more for less', 'tribe-common' );?></h2>
+				<p><?php _e( 'Some text about why bundles are great', 'tribe-common' ); ?></p>
```
Comment (sc0ttkclark) [inline src/admin-views/app-shop.php:119]:
```suggestion
				<h2><?php esc_html_e( 'Get more for less', 'tribe-common' ); ?></h2>
				<p><?php esc_html_e( 'Some text about why bundles are great', 'tribe-common' ); ?></p>
```

File: src/Tribe/App_Shop.php [hunk 4]
```diff
@@ -123,32 +123,166 @@ public function is_current_page() {
...
+				(object) [
+					'title' => __( 'Ultimate Bundle', 'tribe-common' ),
+					'logo' => 'images/logo/bundle-ultimate.svg',
+					'link' => 'https://m.tri.be/somewhere',
+					'discount' => __( 'Save over 40%', 'tribe-common' ),
+					'description' => __( 'All of our premium plugins in one big convenient package]', 'tribe-common' ),
+					'includes' => [
+						'events-calendar-pro',
+						'event-tickets-plus',
+						'virtual-events', /* code review: not true today, confirm this has changed before shipping! */
```
Comment (sc0ttkclark) [inline src/Tribe/App_Shop.php:223]: code review: not true today, confirm this has changed before shipping!

File: src/Tribe/App_Shop.php [hunk 5]
```diff
@@ -123,32 +123,166 @@ public function is_current_page() {
...
+						'events-community',
+						'events-community-tickets',
+						'tribe-filterbar',
+						'event-aggregator',
+						'tribe-eventbrite',
+						'promoter', /* code review: not true today, confirm this has changed before shipping! */
```
Comment (sc0ttkclark) [inline src/Tribe/App_Shop.php:229]: code review: not true today, confirm this has changed before shipping!

File: src/admin-views/app-shop.php [hunk 6]
```diff
@@ -1,95 +1,223 @@
...
+							<div class="headline">
+								<img src="<?php echo esc_url( tribe_resource_url( $bundle->logo, false, null, $main ) ); ?>" alt="<?php esc_attr_e( 'TEC Logo', 'tribe-common' ); ?>" />
+								<h3><a href="<?php echo esc_url( $bundle->link ); ?>" target="_blank"><?php echo esc_html( $bundle->title ); ?></a></h3>
+							</div>
+
+							<p><?php echo esc_html( $bundle->description ); ?></p>
+
+							<div class="cta wide">
+								<a class="button" href="<?php echo esc_url( $bundle->link ); ?>"><?php esc_html_e( 'Save With A Bundle', 'tribe-common' ); ?></a>
+								<span class="discount"><?php echo esc_html( $bundle->discount );?></span>
```
Comment (sc0ttkclark) [inline src/admin-views/app-shop.php:134]:
```suggestion
								<span class="discount"><?php echo esc_html( $bundle->discount ); ?></span>
```

File: src/Tribe/App_Shop.php [hunk 7]
```diff
@@ -123,32 +123,166 @@ public function is_current_page() {
...
+				'event-aggregator' =>         (object) $all_products['event-aggregator'],
+				'event-tickets' =>            (object) $all_products['event-tickets'],
+				'event-tickets-plus' =>       (object) $all_products['event-tickets-plus'],
+				'promoter' =>                 (object) $all_products['promoter'],
+				'tribe-filterbar' =>          (object) $all_products['tribe-filterbar'],
+				'events-community' =>         (object) $all_products['events-community'],
+				'events-community-tickets' => (object) $all_products['events-community-tickets'],
+				'tribe-eventbrite' =>         (object) $all_products['tribe-eventbrite'],
+				'image-widget-plus' =>        (object) $all_products['image-widget-plus'],
+			];
```
Comment (bordoni) [inline src/Tribe/App_Shop.php:152]:
```suggestion
			$products = array_map( static function ( $value ) {
				return (object) $value;
			 } );
```

File: src/Tribe/App_Shop.php [hunk 8]
```diff
@@ -123,32 +123,166 @@ public function is_current_page() {
...
+				(object) [
+					'title' => __( 'Ultimate Bundle', 'tribe-common' ),
+					'logo' => 'images/logo/bundle-ultimate.svg',
+					'link' => 'https://m.tri.be/somewhere', /* code review: fix this */
+					'discount' => __( 'Save over 40%', 'tribe-common' ), /* code review: fix this */
+					'description' => __( 'All of our premium plugins in one big convenient package]', 'tribe-common' ),
+					'includes' => [
+						'events-calendar-pro',
+						'event-tickets-plus',
+						'virtual-events', /* code review: not true today, confirm this has changed before shipping! */
```
Comment (bordoni) [inline src/Tribe/App_Shop.php:223]:
```suggestion
						'events-virtual',
```

File: src/Tribe/App_Shop.php [hunk 9]
```diff
@@ -123,32 +123,166 @@ public function is_current_page() {
...
+				(object) [
+					'title' => __( 'Virtual Event Marketing Bundle', 'tribe-common' ),
+					'logo' => 'images/logo/bundle-virtual-events.svg',
+					'link' => 'https://m.tri.be/somewhere', /* code review: fix this */
+					'discount' => __( 'Save over 40%', 'tribe-common' ), /* code review: fix this */
+					'description' => __( '[description]', 'tribe-common' ), /* code review: fix this */
+					'includes' => [
+						'events-calendar-pro',
+						'event-tickets-plus',
+						'virtual-events',
```
Comment (bordoni) [inline src/Tribe/App_Shop.php:197]:
```suggestion
						'events-virtual',
```

File: src/Tribe/Plugins_API.php [hunk 10]
```diff
@@ -24,121 +24,206 @@ public function get_products() {
...
+					__( 'Slideshow', 'tribe-common' ),
+					__( 'Random Images', 'tribe-common' ),
+				],
+				'image' => 'images/shop/image-widget-plus.jpg',
+				'logo' => 'images/logo/image-widget-plus.svg',
 				'is_installed' => class_exists( 'Tribe__Image__Plus__Main' ),
+				'free' => false,
+				'active_installs' => 2500,
+			),
+			'virtual-events' => array(
```
Comment (bordoni) [inline src/Tribe/Plugins_API.php:212]:
```suggestion
			'events-virtual' => array(
```

File: src/admin-views/app-shop.php [hunk 11]
```diff
@@ -1,95 +1,226 @@
...
+				foreach ( $all_products as $status => $some_products ) {
+
+					if ( 'for-sale' == $status ) {
+						?>
+						<h2><?php esc_html_e( 'One calendar. Countless ways to make it your own.', 'tribe-common' ); ?></h2>
+						<p><?php esc_html_e( 'Calendars, ticketing, and powerful WordPress tools to manage your events from start to finish.', 'tribe-common' ); ?></p>
+						<?php
+					}
+					else {
+						?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:47]:
```suggestion
					<?php else: ?>
```

File: src/admin-views/app-shop.php [hunk 12]
```diff
@@ -1,95 +1,226 @@
...
+	</div>
+
+	<div id="tribe-all-solutions" class="tribe-content">
 		<div class="content-wrapper">
 			<div class="addon-grid">
 				<?php
+				foreach ( $all_products as $status => $some_products ) {
+
+					if ( 'for-sale' == $status ) {
+						?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:41]:
```suggestion
				<?php foreach ( $all_products as $status => $some_products ) : ?>
					<?php if ( 'for-sale' == $status ) : ?>
```

File: src/admin-views/app-shop.php [hunk 13]
```diff
@@ -1,95 +1,226 @@
...
+	</div>
 
+	<div id="tribe-extensions" class="tribe-content">
+		<div class="content-wrapper">
+			<div class="addon-grid">
+				<h2><?php esc_html_e( 'Extensions Headline', 'tribe-common' ); ?></h2>
+				<p><?php esc_html_e( 'Some text about why extensions are great', 'tribe-common' ); ?></p>
+				<?php
+				foreach ( $extensions as $extension ) {
+					?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:184]:
```suggestion
				<?php foreach ( $extensions as $extension ) : ?>
```

File: src/admin-views/app-shop.php [hunk 14]
```diff
@@ -1,95 +1,226 @@
...
+							<a href="<?php echo esc_url( $extension->link ); ?>" target="_blank"><img src="<?php echo esc_url( tribe_resource_url( $extension->image, false, null, $main ) ); ?>" /></a>
+						</div>
+
+						<p><?php echo esc_html( $extension->description ); ?></p>
+
+						<a class="button" href="<?php echo esc_url( $extension->link ); ?>"><?php esc_html_e( 'Download', 'tribe-common' ); ?></a>
+					</div>
 					<?php
 				}
 				?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:201]:
```suggestion
					<?php endforeach; ?>
```

File: src/admin-views/app-shop.php [hunk 15]
```diff
@@ -1,95 +1,226 @@
...
+									$product = $products[ $product_key ];
+
 									?>
-									<p><strong><?php esc_html_e( 'Requires:', 'tribe-common' );?></strong> <?php echo esc_html( $product->requires ); ?></p>
+									<li><img src="<?php echo esc_url( tribe_resource_url( $product->logo, false, null, $main ) ); ?>" alt="<?php esc_attr_e( 'TEC Logo', 'tribe-common' ); ?>" /><span><?php echo esc_html( $product->title ); ?></span></li>
 									<?php
+
+									if ( $i == 4 ) {
+										// if there are 5 products included, then we need 2 lists
+										?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:155]:
```suggestion
									<?php if ( $i == 4 ) : // if there are 5 products included, then we need 2 lists ?>
```

File: src/admin-views/app-shop.php [hunk 16]
```diff
@@ -1,95 +1,226 @@
...
+										</ul><ul class="second">
+										<?php
+									}
 								}
 								?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:160]:
```suggestion
										<?php endif; ?>
								<?php endforeach ?>
```

File: src/admin-views/app-shop.php [hunk 17]
```diff
@@ -1,95 +1,226 @@
...
-								<p><?php echo $product->description; ?></p>
+						<div class="includes">
+							<h4><?php _e( 'Includes', 'tribe-common' ); ?></h4>
+							<ul>
 								<?php
-								if ( isset( $product->requires ) ) {
+								foreach ( $bundle->includes as $i => $product_key ) {
+									// get $product object
+									$product = $products[ $product_key ];
+
```
Comment (bordoni) [inline src/admin-views/app-shop.php:148]:
```suggestion
								foreach ( $bundle->includes as $i => $product_key ) :
									// get $product object
									$product = $products[ $product_key ];

```

File: src/admin-views/app-shop.php [hunk 18]
```diff
@@ -1,95 +1,226 @@
...
 									?>
-									<p><strong><?php esc_html_e( 'Requires:', 'tribe-common' );?></strong> <?php echo esc_html( $product->requires ); ?></p>
+									<li><img src="<?php echo esc_url( tribe_resource_url( $product->logo, false, null, $main ) ); ?>" alt="<?php esc_attr_e( 'TEC Logo', 'tribe-common' ); ?>" /><span><?php echo esc_html( $product->title ); ?></span></li>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:150]:
```suggestion
									<li>
										<img src="<?php echo esc_url( tribe_resource_url( $product->logo, false, null, $main ) ); ?>" alt="<?php esc_attr_e( 'TEC Logo', 'tribe-common' ); ?>" />
										<span><?php echo esc_html( $product->title ); ?></span>
									</li>
```

File: src/admin-views/app-shop.php [hunk 19]
```diff
@@ -1,95 +1,226 @@
...
 
-				foreach ( $products as $i => $product ) {
+	<div id="tribe-bundles" class="tribe-content">
+		<div class="content-wrapper">
+			<div class="addon-grid">
+				<h2><?php esc_html_e( 'Get more for less', 'tribe-common' ); ?></h2>
+				<p><?php esc_html_e( 'Some text about why bundles are great', 'tribe-common' ); ?></p>
+				<?php
+				foreach ( $bundles as $bundle ) {
 					?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:125]:
```suggestion
				<?php foreach ( $bundles as $bundle ) : ?>
```

File: src/admin-views/app-shop.php [hunk 20]
```diff
@@ -1,95 +1,226 @@
...
-							<a class="button <?php esc_attr_e( $button_class ); ?>" href="<?php echo esc_url( $product->link ); ?>"><?php echo $button_label; // escaped above ?></a>
+						<div class="cta narrow">
+							<a class="button" href="<?php echo esc_url( $bundle->link ); ?>"><?php esc_html_e( 'Save With A Bundle', 'tribe-common' ); ?></a>
+							<span class="discount"><?php echo esc_html( $bundle->discount ); ?></span>
 						</div>
+
 					</div>
+					<?php
+				}
+				?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:172]:
```suggestion
				<?php endforeach; ?>
```

File: src/admin-views/app-shop.php [hunk 21]
```diff
@@ -1,95 +1,226 @@
...
+								<img src="<?php echo esc_url( tribe_resource_url( $product->logo, false, null, $main ) ); ?>" alt="<?php esc_attr_e( 'TEC Logo', 'tribe-common' ); ?>" />
+								<h3 <?php echo ( 'installed' == $status || $product->free ) ? 'class="has-pill"' : ''; ?>><a href="<?php echo esc_url( $product->link ); ?>" target="_blank"><?php echo esc_html( $product->title ); ?></a></h3>
+
+								<?php
+								if ( 'installed' == $status ) {
+									?>
+									<span class="pill active"><?php esc_html_e( 'Active', 'tribe-common' ); ?></span>
+									<?php
+								} elseif ( $product->free ) {
+									?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:67]:
```suggestion
								<?php elseif ( $product->free ) : ?>
```

File: src/admin-views/app-shop.php [hunk 22]
```diff
@@ -1,95 +1,226 @@
...
+									<span class="pill free"><?php esc_html_e( 'FREE', 'tribe-common' ); ?></span>
+									<?php
+								} ?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:70]:
```suggestion
								<?php endif; ?>
```

File: src/admin-views/app-shop.php [hunk 23]
```diff
@@ -1,95 +1,226 @@
...
+					else {
+						?>
+						<h2 class="already-installed"><?php esc_html_e( 'Already Installed', 'tribe-common' ); ?></h2>
+						<?php
+					}
+					?>
+
+					<?php
+					foreach ( $some_products as $product ) {
+						?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:55]:
```suggestion
					<?php foreach ( $some_products as $product ) : ?>
```

File: src/admin-views/app-shop.php [hunk 24]
```diff
@@ -1,95 +1,226 @@
...
+						<h2><?php esc_html_e( 'One calendar. Countless ways to make it your own.', 'tribe-common' ); ?></h2>
+						<p><?php esc_html_e( 'Calendars, ticketing, and powerful WordPress tools to manage your events from start to finish.', 'tribe-common' ); ?></p>
+						<?php
+					}
+					else {
+						?>
+						<h2 class="already-installed"><?php esc_html_e( 'Already Installed', 'tribe-common' ); ?></h2>
+						<?php
+					}
+					?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:51]:
```suggestion
					<?php endif; ?>
```

File: src/admin-views/app-shop.php [hunk 25]
```diff
@@ -1,95 +1,226 @@
...
+					foreach ( $some_products as $product ) {
+						?>
+						<div class="tribe-addon">
+							<div class="headline">
+								<img src="<?php echo esc_url( tribe_resource_url( $product->logo, false, null, $main ) ); ?>" alt="<?php esc_attr_e( 'TEC Logo', 'tribe-common' ); ?>" />
+								<h3 <?php echo ( 'installed' == $status || $product->free ) ? 'class="has-pill"' : ''; ?>><a href="<?php echo esc_url( $product->link ); ?>" target="_blank"><?php echo esc_html( $product->title ); ?></a></h3>
+
+								<?php
+								if ( 'installed' == $status ) {
+									?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:63]:
```suggestion
								<?php if ( 'installed' == $status ) : ?>
```

File: src/admin-views/app-shop.php [hunk 26]
```diff
@@ -1,95 +1,226 @@
...
+								<p><?php echo esc_html( $product->description ); ?></p>
+							</div>
 
-					case 2:
-						$wide_indexes = array( 0, 1 );
-						break;
+							<ul class="features">
+								<?php
+								foreach( $product->features as $feature ) {
+									?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:84]:
```suggestion
								<?php foreach( $product->features as $feature ) : ?>
```

File: src/admin-views/app-shop.php [hunk 27]
```diff
@@ -1,95 +1,226 @@
...
+									<li>
+										<span class="check">
+											<svg fill="none" height="12" viewBox="0 0 16 12" width="16" xmlns="http://www.w3.org/2000/svg"><path clip-rule="evenodd" d="m13.7357.374803-8.40784 8.402337-3.06361-3.06158c-.52424-.506-1.357557-.49877-1.872924.01626s-.522608 1.34779-.016275 1.87169l4.008209 4.00559c.52173.5212 1.36747.5212 1.8892 0l9.35244-9.34634c.5064-.5239.4991-1.356665-.0162-1.871692-.5154-.515027-1.3487-.522264-1.873-.016265z" fill="#3d54ff" fill-rule="evenodd"/></svg>
+										</span><span class="feature-text">
+											<?php echo esc_html( $feature ); ?>
+										</span>
+									</li>
+									<?php
+								};
+								?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:94]:
```suggestion
								<?php endforeach; ?>
```

File: src/admin-views/app-shop.php [hunk 28]
```diff
@@ -1,95 +1,226 @@
...
+							</ul>
 
-					case 5:
-						$wide_indexes = array( 0, 4 );
-						break;
+							<?php
+							if ( 'installed' == $status ) {
+								?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:99]:
```suggestion
							<?php if ( 'installed' == $status ) : ?>
```

File: src/admin-views/app-shop.php [hunk 29]
```diff
@@ -1,95 +1,226 @@
...
+								<a class="button" href="https://m.tri.be/1aiz"><?php esc_html_e( 'Manage', 'tribe-common' ); ?></a>
+								<?php
+							} else {
+								?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:103]:
```suggestion
							<?php else : ?>
```

File: src/admin-views/app-shop.php [hunk 30]
```diff
@@ -1,95 +1,226 @@
...
+								<a class="button" href="<?php echo esc_url( $product->link ); ?>"><?php esc_html_e( 'Learn More', 'tribe-common' ); ?></a>
+								<?php
+							}
+							?>
```
Comment (bordoni) [inline src/admin-views/app-shop.php:107]:
```suggestion
							<?php endif; ?>
```

File: src/Tribe/App_Shop.php (hunk 7 above)
Comment (zbtirrell) [reply src/Tribe/App_Shop.php:152]: nope. we want to specifically control the order these are displayed in.

Comment (bordoni) [reply src/Tribe/App_Shop.php:152]: Ahhh I see!

**Pull request merged**